Sources/APNS/APNSContainers.swift

```diff
@@ -0,0 +1,88 @@
+import Vapor
+import APNSwift
+
+public class APNSContainers {
+    public struct ID: Hashable, Codable {
+        public let string: String
+        public init(string: String) {
+            self.string = string
+        }
+    }
+
+    public final class Container {
+        public let configuration: APNSConfiguration
+        public let client: APNSClient
+        
+        init(configuration: APNSConfiguration, client: APNSClient) {
+            self.configuration = configuration
+            self.client = client
+        }
+    }
+
+    private var containers: [ID: Container]
+    private var defaultID: ID?
+    private var lock: Lock
+
+    init() {
+        self.containers = [:]
+        self.lock = .init()
+    }
+}
+
+extension APNSContainers {
+
+    public func use(
+        _ config: APNSConfiguration,
+        as id: ID,
+        isDefault: Bool? = nil
+    ) {
+        self.lock.lock()
```

Comment:
@kylebrowning Wanna take advantage of using an actor here instead to protect mutable state. Makes it easier to understand without the ceremony.

Reply:
I can't until vapor supports that with ```public struct Storage {```

Reply:
Ah gotcha.